### PR TITLE
Makes docker build faster.

### PR DESCRIPTION
**Pull request by hydroid7**
The new dockerfile first pulls the dependecies and in the next step 
builds the binary.

**- Summary**
Minor change to the docker builder stage. It was painful to download all deps on a small code change.

**- Test plan**
n. a.

**- Description for the changelog**
Faster Docker builds.

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by mraerino** [comment]
can you rebase to get CI green?

File: Dockerfile [hunk 1]
```diff
@@ -6,10 +6,16 @@ ENV GOOS=linux
 RUN apk add --no-cache make git
 
 WORKDIR /go/src/github.com/netlify/gotrue
-COPY . /go/src/github.com/netlify/gotrue
 
-RUN make deps build
+# Pulling dependencies
+COPY ./Makefile .
+COPY ./go.mod .
+COPY ./go.sum .
```
Comment (mraerino) [inline Dockerfile:13]: you can do this in one instruction

**Review by mraerino — COMMENTED** [review]
looks good. one nit. can approve once rebased

File: Dockerfile (hunk 1 above)
Comment (hydroid7) [reply Dockerfile]: Good hint. Did it.

**Review by mraerino — APPROVED** [review]
thanks for the optimization! 🚀

**Pull request merged**